parsing/lexer.mli

```diff
@@ -24,6 +24,7 @@ type error =
   | Unterminated_string_in_comment of Location.t * Location.t
   | Keyword_as_label of string
   | Literal_overflow of string
+  | Literal_malformed of string
```

Comment:
True. This was only to get a common prefix with `Literal_overflow`.
`Invalid_literal` seems better anyway.